girder/api/v1/resource.py

```diff
@@ -139,6 +140,96 @@ def _getResourceModel(self, kind, funcName=None):
+        :returns: the child resource
+        """
+
+        filterObject = {'name': token, 'parentId': parent['_id']}
+
+        # look for a folder
+        candidateChild = self.model('folder').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'folder'
+
+        # if folder not found, look for an item
+        filterObject['folderId'] = filterObject.pop('parentId')
+        candidateChild = self.model('item').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'item'
+
+        # if item not found, look for a file
+        filterObject['itemId'] = filterObject.pop('folderId')
+        candidateChild = self.model('file').findOne(filterObject)
+        if candidateChild is not None:
+            return candidateChild, 'file'
+
+        # if no folder, item, or file matches, give up
+        raise RestException('Child resource not found: {}({})->{}'.format(
+            parentType, parent.get('name', parent.get('_id')), token))
+
+    def _lookupPath(self, path, user=None):
+        """
+        Find a particular resource by path or throw an exception.  The given
+        user must have read access to all resources featured in the path.
+        :param path: the path of the object to find
+        :param user: the user to search as
+        :returns: the resource.
+        """
+
+        pathArray = [token for token in path.split('/') if token]
+        model = pathArray[0]
+
+        if user is None:
+            user = self.getCurrentUser()
+
+        parent = None
+        if model == 'user':
+            username = pathArray[1]
+            parent = self.model('user').findOne({'login': username})
+
+            if parent is None:
+                raise RestException('User not found: {}'.format(username))
+
+        elif model == 'collection':
+            collectionName = pathArray[1]
+            parent = self.model('collection').findOne({'name': collectionName})
+
+            if parent is None:
+                raise RestException(
+                    'Collection not found: {}'.format(collectionName))
+
+        else:
+            raise RestException('Invalid path format')
+
+        try:
+            document = parent
+            self.model(model).requireAccess(document, user)
+            for token in pathArray[2:]:
+                document, model = self._lookupToken(token, model, document)
+                self.model(model).requireAccess(document, user)
+        except RestException:
+            raise RestException('Path not found: {}'.format(path))
+
+        result = self.model(model).filter(document, user)
+        return result
+
+    @access.public
+    def stat(self, params):
+        self.requireParams('path', params)
+        return self._lookupPath(params['path'], self.getCurrentUser())
+
+    stat.description = (
+        Description('Get any resource by girder path.')
+        .param('path', 'The path of the resource.')
```

Comment:
Pedantic word nerd here - I think "lookup" is strictly a noun and the description should say "look up".  Sorry, for some reason lookup-vs-"look up" really bugs me :eyeglasses: